app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ReportsController < ApplicationController
+  def index
+    if current_instructor
+      @reports = Report.all
+    else
+      @reports = Report.where(student_id: current_student.id)
+    end
+  end
+
+  def new
+    authenticate_instructor!
```

Comment:
Ah, I see now that this controller is not authenticated in the routes now. Carry on.